File: node/zkgroup/receipts/ReceiptCredentialRequest.ts
//
// Copyright 2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray';
import NativeImpl from '../../NativeImpl';

export default class ReceiptCredentialRequest extends ByteArray {

    static SIZE = 97;

    constructor(contents: Buffer) {
        super(contents, ReceiptCredentialRequest.SIZE, true);
        NativeImpl.ReceiptCredentialRequest_CheckValidContents(contents);
    }
}

File: java/shared/java/org/signal/zkgroup/receipts/ReceiptCredentialRequest.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.receipts;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class ReceiptCredentialRequest extends ByteArray {
  public ReceiptCredentialRequest(byte[] contents) throws InvalidInputException {
    super(contents);
    Native.ReceiptCredentialRequest_CheckValidContents(contents);
  }
}
